### PR TITLE
Merge design

### PR DESCRIPTION
**Pull request by clavin-xlnx**
Adds a new basic capability to RapidWright to merge two or more designs into a single implementation.  Provides an abstract class with an interface that allows the user to customize the decisions made during the merging process.

### CODE REVIEW

File: src/com/xilinx/rapidwright/design/merge/AbstractDesignMerger.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package com.xilinx.rapidwright.design.merge;
+
+import com.xilinx.rapidwright.design.Net;
+import com.xilinx.rapidwright.design.SiteInst;
+import com.xilinx.rapidwright.edif.EDIFCellInst;
+import com.xilinx.rapidwright.edif.EDIFNet;
+import com.xilinx.rapidwright.edif.EDIFPort;
+
```
Comment (eddieh-xlnx) [inline src/com/xilinx/rapidwright/design/merge/AbstractDesignMerger.java:8]: One liner for javadoc?

File: src/com/xilinx/rapidwright/design/merge/DefaultDesignMerger.java [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+import com.xilinx.rapidwright.design.Net;
+import com.xilinx.rapidwright.design.SiteInst;
+import com.xilinx.rapidwright.device.BELPin;
+import com.xilinx.rapidwright.device.PIP;
+import com.xilinx.rapidwright.edif.EDIFCellInst;
+import com.xilinx.rapidwright.edif.EDIFHierCellInst;
+import com.xilinx.rapidwright.edif.EDIFNet;
+import com.xilinx.rapidwright.edif.EDIFPort;
+import com.xilinx.rapidwright.edif.EDIFPortInst;
+
```
Comment (eddieh-xlnx) [inline src/com/xilinx/rapidwright/design/merge/DefaultDesignMerger.java:20]: One liner for javadoc?

File: src/com/xilinx/rapidwright/design/merge/MergeOptions.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
+package com.xilinx.rapidwright.design.merge;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import com.xilinx.rapidwright.design.Net;
+
```
Comment (eddieh-xlnx) [inline src/com/xilinx/rapidwright/design/merge/MergeOptions.java:8]: One liner for javadoc?

File: test/com/xilinx/rapidwright/design/merge/TestMergeDesigns.java [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+                                top1.getNet(reset).getPortInsts().size() - 1);
+        
+        Assertions.assertEquals(top.getCellInsts().size(), countMergedInsts(top0, top1));
+        Assertions.assertEquals(top.getNets().size(), countMergedNets(top0, top1));
+        Assertions.assertEquals(top.getPorts().size(), countMergedPorts(top0, top1));
+        Assertions.assertEquals(merged.getSiteInsts().size(), countMergedSiteInsts(design0, design1));
+    }
+    
+    @ParameterizedTest
+    @ValueSource(strings = {"False", "True"}) // Doesn't support boolean :(
```
Comment (eddieh-xlnx) [inline test/com/xilinx/rapidwright/design/merge/TestMergeDesigns.java:171]: Did you try this? https://stackoverflow.com/a/59958483

**Review by eddieh-xlnx — APPROVED** [review]

File: test/com/xilinx/rapidwright/design/merge/TestMergeDesigns.java (hunk 4 above)
Comment (clavin-xlnx) [reply test/com/xilinx/rapidwright/design/merge/TestMergeDesigns.java]: Huh, I guess some dude on the Internet was wrong.

**Pull request merged**